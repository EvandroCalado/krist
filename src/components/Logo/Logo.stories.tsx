import { Meta, StoryFn } from '@storybook/react';

import { Logo } from '.';

export default {
  title: 'Components/Logo',
  component: Logo,
} as Meta;

export const Default: StoryFn = () => <Logo />;
