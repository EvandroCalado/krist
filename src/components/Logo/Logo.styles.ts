import styled from 'styled-components';

export const Logo = styled.img`
  width: 143px;
  height: 59px;
  object-fit: contain;
`;
