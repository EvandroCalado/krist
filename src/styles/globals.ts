import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  html {
    font-size: 62.5%;
    scroll-behavior: smooth;
  }

  body {
    font-size: 1.6rem;
  }

  a {
    text-decoration: none;
  }

  ol, ul {
    list-style: none;
  }
`;
