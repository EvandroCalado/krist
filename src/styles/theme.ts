export default {
  colors: {
    primary: '#1d4ed8',
    secondary: '#dc2626',
    warning: '#e11d48',
    success: '#16a34a',
    info: '#3b82f6',
    white: '#ffffff',
    black: '#000000',
    background: '#f5f5f5',
    text: '#262626',
  },
  font: {
    family: {
      primary: "'Inter', sans-serif;",
    },
    sizes: {
      xs: '.8rem',
      sm: '1.4rem',
      md: '2.4rem',
      lg: '3.2rem',
      xl: '4rem',
      '2xl': '4.8rem',
      '3xl': '5.6rem',
      '4xl': '6.4rem',
      '5xl': '7.2rem',
      '6xl': '8.0rem',
      '7xl': '9.6rem',
    },
  },
  media: {
    lteOrEqSmallest: '(max-width: 320px)',
    lteOrEqSmall: '(max-width: 576px)',
    lteOrEqMedium: '(max-width: 768px)',
    lteOrEqLarge: '(max-width: 992px)',
    gteOrEqSmallest: '(min-width: 320px)',
    gteOrEqSmall: '(min-width: 576px)',
    gteOrEqMedium: '(min-width: 768px)',
    gteOrEqLarge: '(min-width: 992px)',
  },
  spacings: {
    xs: '0.5rem',
    sm: '1rem',
    md: '2rem',
    lg: '3rem',
    xl: '4rem',
    '2xl': '5rem',
    '3xl': '6rem',
    '4xl': '7rem',
    '5xl': '8rem',
    '6xl': '9rem',
    '7xl': '10rem',
  },
  frameSizes: {
    xs: '8rem',
    sm: '12rem',
    md: '24rem',
    lg: '32rem',
    xl: '40rem',
    '2xl': '56rem',
    '3xl': '64rem',
    '4xl': '80rem',
    '5xl': '104rem',
    '6xl': '120rem',
    '7xl': '134rem',
  },
  transitions: {
    fastest: 'all 100ms ease-in-out',
    faster: 'all 200ms ease-in-out',
    fast: 'all 300ms ease-in-out',
    normal: 'all 500ms ease-in-out',
    slow: 'all 600ms ease-in-out',
    slower: 'all 800ms ease-in-out',
    slowest: 'all 1s ease-in-out',
  },
  layers: {
    normal: 0,
    layer1: 1,
    layer2: 2,
    layer3: 3,
    layer4: 4,
    layer5: 5,
    layer6: 6,
    layer7: 7,
    layer8: 8,
    layer9: 9,
    aboveAll: 10,
  },
} as const;
