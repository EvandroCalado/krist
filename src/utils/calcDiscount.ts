export const calcDiscount = (value: number, discount: number) => {
  const valueWithDiscount = (value * (100 - discount)) / 100;

  if (discount === 0) return value;

  return valueWithDiscount.toFixed(2);
};
