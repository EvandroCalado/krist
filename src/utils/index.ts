export * from './calcDiscount';
export * from './custom-fetch';
export * from './formatDate';
export * from './linksProfileMenu';
export * from './test-utils';
