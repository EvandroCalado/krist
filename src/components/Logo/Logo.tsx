import { Link, useLoaderData } from 'react-router-dom';

import { StrapiConfigType } from 'types/strapi-config-type';

import * as S from './Logo.styles';

export const Logo = () => {
  const { config } = useLoaderData() as { config: StrapiConfigType };

  const { black } = config.data.attributes.logo;

  return (
    <Link to="/">
      <S.Logo src={black.data.attributes.url} alt="logo" />
    </Link>
  );
};
