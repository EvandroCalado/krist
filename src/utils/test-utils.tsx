import { render, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from 'styles/theme';

type CustomRenderProps = Omit<RenderOptions, 'queries'>;

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {},
) => render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions);

export * from '@testing-library/react';
export { customRender as render };
