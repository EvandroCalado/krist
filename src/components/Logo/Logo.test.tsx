import { screen } from '@testing-library/react';

import { customRender } from 'utils/test-utils';

import { Logo } from './Logo.styles';

describe('Logo', () => {
  it('should render default', () => {
    customRender(<Logo />);

    const logo = screen.getByRole('img');

    expect(logo).toBeInTheDocument();
    expect(logo).toHaveStyle({
      width: '143px',
      height: '59px',
    });
  });

  it('should render a snapshot', () => {
    const { container } = customRender(<Logo />);

    expect(container).toMatchSnapshot();
  });
});
