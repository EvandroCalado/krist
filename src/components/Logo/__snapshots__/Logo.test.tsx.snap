// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render a snapshot 1`] = `
.c0 {
  width: 143px;
  height: 59px;
  object-fit: contain;
}

@media (max-width: 768px) {
  .c0 {
    width: 100px;
  }
}

@media (max-width: 576px) {
  .c0 {
    width: 70px;
  }
}

<div>
  <a
    href="/"
  >
    <img
      alt="logo"
      class="c0"
      src="image-white"
    />
  </a>
  ,
</div>
`;
